docs(icons): update icon fonts page (#13194)
vuetifyjs/vuetify: packages/docs/src/pages/en/features/icon-fonts.md

# Icon Fonts

Vuetify comes with support for Material Design Icons, Material Icons, Font Awesome 4 and Font Awesome 5. By default, applications will use [Material Design Icons](https://materialdesignicons.com/).

<entry-ad />

## Usage

In order to change your font library, you will need to either import one of our pre-defined icon sets, or provide your own.

```js
// src/plugins/vuetify.js

import { createVuetify } from 'vuetify'
import { aliases, mdi } from 'vuetify/src/iconsets/mdi'

export default new createVuetify({
  icons: {
    defaultSet: 'mdi',
    aliases,
    sets: {
      mdi,
    }
  },
})
```

```html
<template>
  <v-icon icon="mdi-home" />
</template>
```

In the above example we are importing the provided `mdi` icon set and its aliases. The aliases are utilized to know which icon in a set is to be used when representing common actions such as close or expand in our internal components.

<alert type="info">

  While it is still possible to supply the icon value throught the default slot in Vuetify 3.0 (`<v-icon>mdi-home</v-icon>`), we recommend that you use the `icon` prop instead.

</alert>

## Multiple icon sets

> You can of course use multiple icon sets at the same time. In the example below, the `fa` set is used as the default set, with `mdi` available as an extra set.

Reviewer: ```suggestion
Out of the box, Vuetify supports the use of multiple *different* icon sets at the same time. The following example demonstrates how to change the default icon font to Font Awesome (`fa`) while still maintaining access to the original Material Design Icons (`mdi`):
```

```js
// src/plugins/vuetify.js

import { createVuetify } from 'vuetify'
import { aliases, fa } from 'vuetify/src/iconsets/fa'
import { mdi } from 'vuetify/src/iconsets/mdi'

export default new createVuetify({
  icons: {
    defaultSet: 'fa',
    aliases,
    sets: {
      fa,
      mdi,
    }
  },
})
```

```html
<template>

>   <v-icon icon="fas fa-plus" /> // This will render a FontAwesome icon

Reviewer: ```suggestion
  <v-icon icon="fas fa-plus" /> // This renders a FontAwesome icon
```

>   <v-icon icon="mdi:mdi-minus" /> // This will render an MDI icon

Reviewer: ```suggestion
  <v-icon icon="mdi:mdi-minus" /> // This renders a MDI icon
```

</template>
```

> Note the `mdi:` prefix used for the second icon. It is necessary to provide the correct prefix when you want to render icons not from the default icon set.

Reviewer: ```suggestion
<alert type="info">

  Note the `mdi:` prefix used in the *second* icon. It is necessary to provide the **correct** prefix when rendering icons that are **not** part of the default set.

</alert>  
```

# Icon Fonts

Out of the box, Vuetify supports 4 popular icon font libraries—[Material Design Icons](https://materialdesignicons.com/), [Material Icons](https://material.io/resources/icons/?style=baseline), [Font Awesome 4](https://fontawesome.com/v4.7.0/) and [Font Awesome 5](https://fontawesome.com/).

<entry-ad />

## Usage

To change your font library, import one of the pre-defined icon sets or provide your own.

```js
// src/plugins/vuetify.js

import { createVuetify } from 'vuetify'

> import { aliases, mdi } from 'vuetify/src/iconsets/mdi'

Reviewer 2: lib

> export default new createVuetify({

Reviewer 2: no new

</template>
```

In the above examples we import the default `mdi` icon set and its corresponding aliases. These aliases reference commonly used types of icons that are utilized by Vuetify components.

<alert type="info">

  While it is still possible to supply the icon value through the default slot in Vuetify 3.0 (`<v-icon>mdi-home</v-icon>`), we recommend using the `icon` prop instead.

</alert>

## Multiple icon sets

Out of the box, Vuetify supports the use of multiple *different* icon sets at the same time. The following example demonstrates how to change the default icon font to Font Awesome (`fa`) while still maintaining access to the original Material Design Icons (`mdi`):

```js
// src/plugins/vuetify.js

import { createVuetify } from 'vuetify'
import { aliases, fa } from 'vuetify/lib/iconsets/fa'
import { mdi } from 'vuetify/lib/iconsets/mdi'

export default createVuetify({
  icons: {
    defaultSet: 'fa',
    aliases,
    sets: {
      fa,
      mdi,
    }
  },
})
```

```html
<template>
  <v-icon icon="fas fa-plus" /> // This renders a FontAwesome icon
  <v-icon icon="mdi:mdi-minus" /> // This renders a MDI icon
</template>
```

<alert type="info">

>   Note the `mdi:` prefix used in the *second* icon. It is necessary to provide the **correct** prefix when rendering icons that are **not** part of the default set.

Reviewer 2: This seems kinda confusing, maybe say "It is not necessary to provide a prefix for icons from the `defaultSet`"

<alert type="info">

>   It is not necessary to provide a prefix (such as `mdi:`) for icons from the default set

Reviewer: ```suggestion
  Icons from the default set can omit the library prefix; e.g. **icon="mdi:close"** → **icon="close"**.
```

Author: What about the current wording don't you like?

### Material Design Icons

> This is the default icon set for Vuetify. You can include it through a CDN:

Reviewer: ```suggestion
This is the default icon set used by Vuetify. It supports local installation with a build process or a CDN link. The following shows how to add the CDN link to your `index.html`:
```

// src/plugins/vuetify.js

import '@mdi/font/css/materialdesignicons.css' // Ensure you are using css-loader
import { createVuetify } from 'vuetify/lib'

export default new createVuetify({
  icons: {
    defaultSet: 'mdi', // This is already the default value - only for display purposes
  },
})
```

> 

Reviewer: ```suggestion
<alert type="error">

  **DO NOT** use a CDN link without specifying a package *version*. Failure to do so can result in unexpected changes to your application with new releases.
 
</alert>

```

Use this tool to search for any Material Design Icons and copy them to your clipboard by clicking the item.

<icon-list />

### Material Design Icons - JS SVG

Use the SVG paths as designated in [@mdi/js](https://www.npmjs.com/package/@mdi/js). This is the recommended installation when optimizing your application for production.

```bash
$ yarn add @mdi/js -D
// OR
$ npm install @mdi/js -D
```

```js
// src/plugins/vuetify.js

import { createVuetify } from 'vuetify/lib'
import { aliases, mdi } from 'vuetify/src/iconsets/mdi-svg'

export default new createVuetify({
  icons: {
    iconfont: 'mdi',
    aliases,
    sets: {
      mdi,
    },
  },
})
```

> You can choose to import only the icons that you need, giving you a much smaller bundle size.

Reviewer: ```suggestion
To reduce bundle size, only import the icons that you need. The following example shows how to use an imported icon within a `.vue` template:
```

```html
<template>
  <v-icon :icon="mdiAccount" />
</template>

<script>
  import { mdiAccount } from '@mdi/js'

  export default {
    data: () => ({
      mdiAccount
    }),
  }
</script>
```

### Material Icons

> For projects without a build process, it is recommended to import the icons from CDN

Reviewer: ```suggestion
For projects without a build process, it is recommended to import the icons from CDN.
```

// src/plugins/vuetify.js

import '@mdi/font/css/materialdesignicons.css' // Ensure you are using css-loader
import { createVuetify } from 'vuetify/lib'

export default createVuetify({
  icons: {
    defaultSet: 'mdi', // This is already the default value - only for display purposes
  },
})
```
<alert type="error">

  **DO NOT** use a CDN link without specifying a package *version*. Failure to do so can result in unexpected changes to your application with new releases.

</alert>

Use this tool to search for any Material Design Icons and copy them to your clipboard by clicking the item.

<icon-list />

### Material Design Icons - JS SVG

Use the SVG paths as designated in [@mdi/js](https://www.npmjs.com/package/@mdi/js). This is the recommended installation when optimizing your application for production.

```bash
$ yarn add @mdi/js -D
// OR
$ npm install @mdi/js -D
```

```js
// src/plugins/vuetify.js

import { createVuetify } from 'vuetify/lib'
import { aliases, mdi } from 'vuetify/lib/iconsets/mdi-svg'

export default createVuetify({
  icons: {

>     iconfont: 'mdi',

Reviewer 2: This doesn't actually allow for arbitrary icons to be used, you still have to import anything not in the preset.

Author: And that is what's shown in the next snippet. Or do you want some text to be extra clear?

export default createVuetify({
  icons: {

>     defaultSet: 'mdi',

Reviewer: Is it bad practice to simply use **default**? We use this convention in `lang` with the **defaultLocale** option, but I'm curious as to your thoughts.

Author: I generally avoid `default` in any scenario because it is a reserved word.

$ npm install material-design-icons-iconfont -D
```

```js
// src/plugins/vuetify.js

import 'material-design-icons-iconfont/dist/material-design-icons.css' // Ensure your project is capable of handling css files
import { createVuetify } from 'vuetify/lib'
import { aliases, md } from 'vuetify/src/iconsets/md'

export default new createVuetify({
  icons: {
    defaultSet: 'md',
    aliases,
    sets: {
      md,
    },
  },
})
```

```html
<template>
  <v-icon icon="home" />
</template>
```

> Some Material Icons are missing by default. For example, `person` and `person_outline` are available, but `visibility_outline` isn't, while `visibility` is. To use the missing icons, replace the existing `<link>` with the following one.

Reviewer: ```suggestion
Some Material Icons are missing by default. For example, `person` and `person_outline` are available, but `visibility_outline` isn't, while `visibility` is. To use the missing icons, replace the existing `<link>` with the following:
```

</template>
```

### Font Awesome SVG Icons

Install the following packages.

```bash
$ yarn add @fortawesome/fontawesome-svg-core @fortawesome/vue-fontawesome @fortawesome/free-solid-svg-icons -D
// or
$ npm install @fortawesome/fontawesome-svg-core @fortawesome/vue-fontawesome @fortawesome/free-solid-svg-icons -D
```

Then register the global `font-awesome-icon` component and use the pre-defined `fa-svg` icon set. If you have access to Font Awesome Pro icons they can be added to the library in the same way.

```js
// src/plugins/vuetify.js

import Vue from 'vue'
import { createVuetify } from 'vuetify'
import { aliases, fa } from 'vuetify/src/iconsets/fa-svg'
import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
import { fas } from '@fortawesome/free-solid-svg-icons'

Vue.component('font-awesome-icon', FontAwesomeIcon) // Register component globally
library.add(fas) // Include needed icons

export default new createVuetify({
  icons: {
    defaultSet: 'fa',
    aliases,
    sets: {
      fa,
    },
  },
})
```

```html
<template>
  <v-icon icon="fas fa-home" />
</template>
```

## Creating a custom icon set

> As previously mentioned you can easily create your own icon set, as shown below. An icon set consists of an object with one property `component` which should be a functional component that receives props of type `IconsProps`.

Reviewer: ```suggestion
An icon set consists of an object with one property `component` which should be a functional component that receives props of type `IconsProps`.
```

> If this new custom icon set is supposed to be your default set, then you should also supply the necessary `aliases` so that Vuetify components know what icons to render.

Reviewer: ```suggestion
In order to use a custom icon set as the framework default, you **must** add the necessary *aliases* that correspond to values used by Vuetify components.

The following snippet is a complete list of all supported icon types:

```

$ npm install material-design-icons-iconfont -D
```

```js
// src/plugins/vuetify.js

import 'material-design-icons-iconfont/dist/material-design-icons.css' // Ensure your project is capable of handling css files
import { createVuetify } from 'vuetify/lib'
import { aliases, md } from 'vuetify/lib/iconsets/md'

export default createVuetify({
  icons: {
    defaultSet: 'md',
    aliases,
    sets: {
      md,
    },
  },
})
```

```html
<template>
  <v-icon icon="home" />
</template>
```

Some Material Icons are missing by default. For example, `person` and `person_outline` are available, but `visibility_outline` isn't, while `visibility` is. To use the missing icons, replace the existing `<link>` with the following:

```html
<link
  rel="stylesheet"
  href="https://fonts.googleapis.com/css?family=Material+Icons|Material+Icons+Outlined|Material+Icons+Two+Tone|Material+Icons+Round|Material+Icons+Sharp"
/>
```

### Font Awesome 5 Icons

> The easiest way to get started with **FontAwesome** is to use a CDN.

Reviewer 2: Add a warning **not** to use the js kits, only SVG and CSS will work with vue.

import { createVuetify } from 'vuetify'
import { aliases, fa } from 'vuetify/lib/iconsets/fa4'

export default createVuetify({
  icons: {
    defaultSet: 'fa',
    aliases,
    sets: {
      fa,
    }
  },
})
```

```html
<template>
  <v-icon icon="fa-check" />
</template>
```

### Font Awesome SVG Icons

Install the following packages.

```bash
$ yarn add @fortawesome/fontawesome-svg-core @fortawesome/vue-fontawesome @fortawesome/free-solid-svg-icons -D
// or
$ npm install @fortawesome/fontawesome-svg-core @fortawesome/vue-fontawesome @fortawesome/free-solid-svg-icons -D
```

Then register the global `font-awesome-icon` component and use the pre-defined `fa-svg` icon set. If you have access to Font Awesome Pro icons they can be added to the library in the same way.

```js
// src/plugins/vuetify.js

import Vue from 'vue'
import { createVuetify } from 'vuetify'
import { aliases, fa } from 'vuetify/lib/iconsets/fa-svg'
import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
import { fas } from '@fortawesome/free-solid-svg-icons'

> Vue.component('font-awesome-icon', FontAwesomeIcon) // Register component globally

Reviewer 2: This is `app.component` now.

</template>
```

<alert type="error">

>   The JavaScript version (`all.js`) of the FontAwesome icons will **not** work with Vue

Reviewer: ```suggestion
  The JavaScript version (`all.js`) of the FontAwesome icons will **NOT** work with Vue
```
